community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Transaction.java

```diff
@@ -19,7 +19,25 @@
  */
 package org.neo4j.internal.kernel.api;
 
-public interface Transaction extends Read, Write
+public interface Transaction
 {
+    long READ_ONLY_TRANSACTION = 0;
```

Comment:
In that case I think the dependency should go in the direction such that `BASE_TX_ID` is defined in terms of this value, so that in order to understand this publicly exposed constant, you don't have to first understand some deeply internal constant.

Reply:
I chose to only comment on this so far. I'm now sure how this will be used, and if we were to change the internal store transaction ids at some point, we could always map ids to match the Kernel API. Let's not overthink this yet.